File: labs/07/NameTester.rb
# NameTester.rb tests class Name and its operations
# Begun by: Prof. Adams, for CS 214 at Calvin College
# Student: Ian Christensen, for CS 214 at Calvin College
# Date: Spring, 2018

require 'test/unit/assertions'
include Test::Unit::Assertions

# Class for a name object
class Name
	def initialize(first, middle, last)
		@first, @middle, @last = first, middle, last
	end

	def fullName()
		@first + ' ' + @middle + ' ' + @last
	end

	def print()
		puts fullName()
		fullName()
	end

	attr_reader :first, :middle, :last
end

# Driver for testing the name class
if __FILE__ == $0
	name = Name.new("John", "Paul", "Jones")
	assert name.first == "John", "first failed"
	assert name.middle == "Paul", "middle failed"
	assert name.last == "Jones", "last failed"
	assert name.fullName == "John Paul Jones", "fullName failed"
	assert name.print == "John Paul Jones", "print failed"
	print "All tests passed!\n"
end

File: labs/07/NameTester.java
/* NameTester.java is a 'driver' to test class Name
 * Begun by: Prof. Adams, for CS 214 at Calvin College
 * Student: Ian Christensen, for CS 214 at Calvin College
 * Date: Spring, 2018
 */

// Driver for testing the name class
public class NameTester {
	public static void main(String[] args) {
		Name aName = new Name("John", "Paul", "Jones");
		assert aName.getFirst().equals("John");
		assert aName.getMiddle().equals("Paul");
		assert aName.getLast().equals("Jones");
		System.out.println(aName);
		assert aName.toString().equals("John Paul Jones");
		System.out.println("All tests passed!");
	}
}

// Class for a name object
class Name {
	private String myFirst, myMiddle, myLast;
	public Name(String first, String middle, String last) {
		myFirst = first;
		myMiddle = middle;
		myLast = last;
	}
	public String getFirst() { return myFirst; }
	public String getMiddle() { return myMiddle; }
	public String getLast() { return myLast; }
	public String toString() { return myFirst + ' ' + myMiddle + ' ' + myLast; }
};
